common/class.tx_dlf_list.php

```diff
@@ -267,12 +268,30 @@ protected function getRecord($element) {
 
                     // Set additional query parameters.
                     $params['start'] = 0;
-                    $params['rows'] = $this->solr->limit;
 
-                    // Set query.
-                    $params['query'] = 'uid:'.tx_dlf_solr::escapeQuery($record['uid']);
+                    // All results are needed, so set the limit unreachable high.
+                    $params['rows'] = 1000000000;
 
-                    // Perform search.
+                    // Take over existing filter queries.
+                    $params['filterquery'] = isset($this->metadata['options']['params']['filterquery']) ? $this->metadata['options']['params']['filterquery'] : array ();
+
+                    // Extend filter query to get all documents with the same uid.
+                    foreach ($params['filterquery'] as $key=>$value) {
+
+                        $params['filterquery'][$key] = array ('query' => $value['query'].' OR toplevel:true');
```

Comment:
Shouldn't this be something like
```suggestion
                        $params['filterquery'][$key]['query'] = $value['query'].' OR toplevel:true';
```
to avoid deleting other array keys besides 'query'?

Reply:
I think we can savely change this. At the moment we don't use any other array keys besides query, but it increases the flexibility for future changes. I think we should change a second occurrence, too.  [tx_dlf_solr](https://github.com/frank-ulrich-weber/goobi-presentation/blob/reworkListView/common/class.tx_dlf_solr.php#L377)